src/Common/FileCache.cpp

```diff
@@ -59,6 +59,24 @@ String FileCache::getPathInLocalCache(const Key & key) const
     return fs::path(cache_base_path) / key_str.substr(0, 3) / key_str;
 }
 
+void FileCache::removeKeyDirectoryIfExists(const Key & key, std::lock_guard<std::mutex> & /* cache_lock */) const
```

Comment:
FYI Sometimes it's possible to use TSA annotations instead of passing lock_guard everywhere